src/praisonai-bot/praisonai_bot/scheduler/_standalone_sender.py

```
@@ -111,17 +136,63 @@ def _post_json(url: str, payload: dict, *, headers: Optional[dict] = None) -> No
+    are retried; a permanent failure (bad token, 4xx) raises on the first
+    attempt so the executor records ``delivery_error`` promptly rather than
+    burning the whole tick on a doomed send.
+    """
+    try:
+        from ..bots._resilience import (
+            BackoffPolicy,
+            compute_backoff,
+            is_recoverable_error,
+            server_retry_after,
+        )
+    except Exception:  # pragma: no cover - defensive: helper ships in-package
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)